python/ray/data/_internal/execution/operators/actor_pool_submitter.py

```diff
@@ -99,40 +254,179 @@ def submit(
         yield from _map_task(fn, *blocks)
 
 
-class ActorPool:
+@dataclass
+class AutoscalingConfig:
+    """Configuration for an autoscaling actor pool."""
+
+    # Minimum number of workers in the actor pool.
+    min_workers: int
+    # Maximum number of workers in the actor pool.
+    max_workers: int
+    # Maximum number of tasks that can be in flight for a single worker.
+    # TODO(Clark): Have this informed by the prefetch_batches configuration, once async
+    # prefetching has been ported to this new actor pool.
+    max_tasks_in_flight: int = 2
+    # Minimum ratio of ready workers to the total number of workers. If the pool is
+    # above this ratio, it will be allowed to be scaled up.
+    # NOTE: This is set to a lower value than the legacy actor compute strategy since we
+    # need to more aggressively scale-up with an event-based autoscaler rather than a
+    # 10ms control loop autoscaler.
```

Comment:
Could we make the policy decision fully independent of the timing? I think the change would be to launch as many actors as needed until should_scale_up() returns False, instead of 1 actor at a time, right?

Reply:
Ah this comment is out-of-date; I had a more involved queue-based + timing-based heuristic that turned this ratio into a fallback. The autoscaling policy shouldn't need to be timing-dependent until we start actually taking time into account (e.g. bundle arrival rates, actor startup time, task execution time, idle timeouts, etc.).

I'll remove this comment and set this ratio back to 0.8!